src/theme.js

```diff
@@ -60,6 +60,22 @@ function getCustomTheme(themeName) {
     return customTheme;
 }
 
+/**
+ * Gets the underlying theme name for the given theme. This is usually the theme or
+ * CSS resource that the theme relies upon to load.
+ * @param {string} theme The theme name to get the base of.
+ * @returns {string} The base theme (typically "light" or "dark").
+ */
+export function getBaseTheme(theme) {
+    if (!theme) return "light";
+    if (theme.startsWith("custom-")) {
```

Comment:
`setTheme` in the same module shares a lot of this code... Any chance to de-duplicate?

Reply:
I'm not sure it's much better... thoughts?
```diff
diff --git a/src/theme.js b/src/theme.js
index 8a15c606d..b346a915b 100644
--- a/src/theme.js
+++ b/src/theme.js
@@ -85,10 +85,9 @@ export function setTheme(theme) {
     if (!theme) {
         theme = SettingsStore.getValue("theme");
     }
-    let stylesheetName = theme;
-    if (theme.startsWith("custom-")) {
+    let stylesheetName = getBaseTheme(theme);
+    if (stylesheetName.endsWith("-custom")) {
         const customTheme = getCustomTheme(theme.substr(7));
-        stylesheetName = customTheme.is_dark ? "dark-custom" : "light-custom";
         setCustomThemeVars(customTheme);
     }
```

Reply:
Hmm, alright, let's go with what you had then! Thanks for checking it out. 😄